### ISSUE TITLE
OS.execute without blocking mode causes zombie processes (defunct)

### ISSUE THREAD

**Issue by rennomarcus** [godotengine/godot#51366]
### Godot version

3.3.2

### System information

Ubuntu 20.04.2 LTS, 64-bit

### Issue description

When executing `OS.execute` with the `blocking` flag as false the process stays in zombie mode. 

![zombie](https://user-images.githubusercontent.com/1592068/128604256-11c00727-ebc8-4073-8091-d6df9350cc2d.png)

Because the process is already 'dead' I cannot kill it, thus remove from the process list. 

### Steps to reproduce

1. run any command in `OS.execute` with the flag `blocking` as false. Example: `var pid = OS.execute('pwd', [], false)`
2. go to your terminal and look for your processes and grep by keyword 'defunct'. Example: `ps cax | grep defunct`

You will see a list of processes executed by godot. If you keep running the command the list keeps growing. When you exit the application/game the processes are removed.

**Comment by qrrk** [comment]
I ran into the same issue today. I start extracting an archive using tar, then wait in a loop checking if this process is still running (using ps, this time in blocking mode) before doing something with the extracted files. Well, with this issue the process never terminates, so that doesn't work. I'll try working around it using threads or maybe multiple nodes and signals, and post the results.

**Comment by qrrk** [comment]
This will do the trick:

```
class_name OSExecWrapper
extends Object


signal process_exited

var _worker: Thread
var output = []
var exit_code = null


func _wrapper(path_and_args: Array) -> void:
	
	exit_code = OS.execute(path_and_args[0], path_and_args[1], true, output)
	emit_signal("process_exited")
	_worker.call_deferred("wait_to_finish")

func execute(path: String, args: PoolStringArray) -> void:
	
	_worker = Thread.new()
	_worker.start(self, "_wrapper", [path, args])
```

Usage example:

```
func _on_Button5_pressed() -> void:
	
	var oew = OSExecWrapper.new()
	print("Starting.")
	oew.execute("sh", ["-c", "sleep 1; ls /tmp"])
	yield(oew, "process_exited")
	print("Finished. Exit code: %s, output:\n%s" % [oew.exit_code, oew.output[0]])
```

**Comment by Ev1lbl0w** [comment]
For 4.0 this has been fixed by #44514
